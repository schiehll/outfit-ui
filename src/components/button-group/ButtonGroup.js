import React, {PropTypes, Children, cloneElement} from 'react'
import suitup from 'suit-up'
import style from './ButtonGroup.style'
import types from '../../config/types'
import removeInvalidProps from '../../utils/removeInvalidProps'

const ButtonGroup = props => {
  const {children, className, styles, ...otherProps} = props

  let classes = ``
  let shouldBeDefault = true

  types.forEach(prop => {
    if (props[prop]) {
      if (shouldBeDefault &&
        prop !== 'default' &&
        types.indexOf(prop) >= 0) {
        shouldBeDefault = false
      }
      classes += ` ${styles[prop]} `
    }
  })

  if (shouldBeDefault) classes += `${styles.default}`
  if (className) classes += ` ${className}`

  const validProps = removeInvalidProps(otherProps)

  const childrenWithProps = Children.map(children, child => {
    return cloneElement(child, otherProps)
  })

  return (
    <div
      className={classes.trim()}
      {...validProps}
    >
      {childrenWithProps}
    </div>
  )
}

ButtonGroup.propTypes = {
  className: PropTypes.string,
  outline: PropTypes.bool,
  round: PropTypes.bool,
  block: PropTypes.bool,
  small: PropTypes.bool,
  big: PropTypes.bool,
  primary: PropTypes.bool,
  success: PropTypes.bool,
  warning: PropTypes.bool,
  danger: PropTypes.bool
}

export default suitup(style)(ButtonGroup)
