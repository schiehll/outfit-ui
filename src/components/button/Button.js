import React, {PropTypes} from 'react'
import suitup from 'suit-up'
import style from './Button.style'
import colors from '../../config/colors'
import sizes from '../../config/sizes'
import types from '../../config/types'
import removeInvalidProps from '../../utils/removeInvalidProps'

const Button = props => {
  const {children, className, styles, ...otherProps} = props

  let classes = ``
  let shouldBeDefault = true

  const possibleProps = colors.concat(sizes).concat(types)
  possibleProps.forEach(prop => {
    if (props[prop]) {
      if (shouldBeDefault &&
        prop !== 'default' &&
        colors.indexOf(prop) >= 0) {
        shouldBeDefault = false
      }
      classes += ` ${styles[prop]} `
    }
  })

  if (shouldBeDefault) classes += `${styles.default}`
  if (className) classes += ` ${className}`

  const validProps = removeInvalidProps(otherProps)

  return (
    <button
      className={classes.trim()}
      {...validProps}
    >
      {children}
    </button>
  )
}

Button.propTypes = {
  className: PropTypes.string,
  outline: PropTypes.bool,
  round: PropTypes.bool,
  block: PropTypes.bool,
  small: PropTypes.bool,
  big: PropTypes.bool,
  primary: PropTypes.bool,
  success: PropTypes.bool,
  warning: PropTypes.bool,
  danger: PropTypes.bool
}

export default suitup(style)(Button)
