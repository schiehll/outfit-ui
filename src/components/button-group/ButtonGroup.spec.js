import React from 'react'
import {render} from 'enzyme'
import {renderToJson} from 'enzyme-to-json'
import {ThemeProvider} from 'suit-up'
import ButtonGroup from './index'
import Button from '../button'

let styles
let theme

beforeAll(() => {
  theme = require('../../theme.example').default

  styles = {
    default: 'default'
  }
})

test('should render just fine', () => {
  const props = {styles}

  const component = render(
    <ThemeProvider theme={theme}>
      <ButtonGroup {...props}>
        <Button>Btn 1</Button>
        <Button>Btn 2</Button>
      </ButtonGroup>
    </ThemeProvider>
  )

  expect(renderToJson(component)).toMatchSnapshot()
})

test('should accept props and pass the relevant ones down', () => {
  let props = {styles, outline: true}
  let component = render(
    <ThemeProvider theme={theme}>
      <ButtonGroup {...props}>
        <Button>Btn 1</Button>
        <Button>Btn 2</Button>
      </ButtonGroup>
    </ThemeProvider>
  )

  expect(renderToJson(component)).toMatchSnapshot()

  props = {styles, round: true}
  component = render(
    <ThemeProvider theme={theme}>
      <ButtonGroup {...props}>
        <Button>Btn 1</Button>
        <Button>Btn 2</Button>
      </ButtonGroup>
    </ThemeProvider>
  )

  expect(renderToJson(component)).toMatchSnapshot()

  props = {styles, round: true, warning: true}
  component = render(
    <ThemeProvider theme={theme}>
      <ButtonGroup {...props}>
        <Button>Btn 1</Button>
        <Button>Btn 2</Button>
      </ButtonGroup>
    </ThemeProvider>
  )

  expect(renderToJson(component)).toMatchSnapshot()
})

test('should accept a className and merge it', () => {
  const props = {styles}

  const component = render(
    <ThemeProvider theme={theme}>
      <ButtonGroup className='someClass' {...props}>
        <Button>Btn 1</Button>
        <Button>Btn 2</Button>
      </ButtonGroup>
    </ThemeProvider>
  )

  expect(renderToJson(component)).toMatchSnapshot()
})
