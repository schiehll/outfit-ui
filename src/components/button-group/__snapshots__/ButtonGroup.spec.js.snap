exports[`test should accept a className and merge it 1`] = `
<div
  class="default_1xgo3 someClass">
  <button
    class="default_1hm0j base_1hm0j">
    Btn 1
  </button>
  <button
    class="default_1hm0j base_1hm0j">
    Btn 2
  </button>
</div>
`;

exports[`test should accept props and pass the relevant ones down 1`] = `
<div
  class="outline_1xgo3">
  <button
    class="outline_1hm0j default_1hm0j base_1hm0j">
    Btn 1
  </button>
  <button
    class="outline_1hm0j default_1hm0j base_1hm0j">
    Btn 2
  </button>
</div>
`;

exports[`test should accept props and pass the relevant ones down 2`] = `
<div
  class="round_1xgo3">
  <button
    class="round_1hm0j default_1hm0j base_1hm0j">
    Btn 1
  </button>
  <button
    class="round_1hm0j default_1hm0j base_1hm0j">
    Btn 2
  </button>
</div>
`;

exports[`test should accept props and pass the relevant ones down 3`] = `
<div
  class="round_1xgo3">
  <button
    class="warning_1hm0j base_1hm0j  round_1hm0j">
    Btn 1
  </button>
  <button
    class="warning_1hm0j base_1hm0j  round_1hm0j">
    Btn 2
  </button>
</div>
`;

exports[`test should render just fine 1`] = `
<div
  class="default_1xgo3">
  <button
    class="default_1hm0j base_1hm0j">
    Btn 1
  </button>
  <button
    class="default_1hm0j base_1hm0j">
    Btn 2
  </button>
</div>
`;
